File: src/eSims.Data/Building/RoomTemplate.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace eSims.Data.Building
{
  public class RoomTemplate
  {
    [Key]
    public int Id { get; set; }

    public string Name { get; set; }

    public double Price { get; set; }

    public string Icon { get; set; }

    //Position

    public int Width { get; set; }
    public int Height { get; set; }

    //Functions

    public int WorkplaceMaxCount { get; set; }

    public int BathroomMaxCount { get; set; }

    public int SmokeMaxCount { get; set; }

    public int KitchenMaxCount { get; set; }
    
    //Extensions

    public List<RoomExtension> RoomExtensions { get; set; }

    public List<WallExtension> WallExtensions { get; set; }

    public bool IsSystemOnly { get; set; }
  }
}

File: src/eSims.Client/scripts/building/room.template.ts
import { RoomExtension } from './room.extension';
import { WallExtension } from './wall.extension';

export class RoomTemplate {
  public id: number;
  public name: string;

  public roomTemplateId: number;

  public price: number;

  public icon: string;

   //Position

  public width: number;
  public height: number;

  //Functions

  public workplaceMaxCount: number;

  public bathroomMaxCount: number;

  public smokeMaxCount: number;

  public kitchenMaxCount: number;

  //Extensions

  public roomExtensions: RoomExtension[];

  public wallExtensions: WallExtension[];

}